File: JavaScriptCodes/05_Valid_Anagram.js
function valid_anagram(s, t) {
    if (s.length != t.length) return false;
    let count = Array(26).fill(0);

    for (let i = 0; i < s.length; i++){
        count[s.charCodeAt(i) - 97]++;
        count[t.charCodeAt(i) - 97]--;
    }

    return count.every(x => x === 0);

}

console.log(valid_anagram("anagram", "margana"))

File: Revision/05_Valid_Anagram.cpp
#include<iostream>
#include<vector>
using namespace std;

bool valid_anagram(string str1, string str2){
    if(str1.length()!=str2.length()){
        return false;
    }
    vector<int> seen(26,0);
    int n = str1.size();
    for (int c = 0; c < n;++c){
        seen[str1[c]-'a']++;
        seen[str2[c]-'a']--;
    }
    for(int c:seen){
        if(c!=0){
            return false;
        }
    }
    return true;
}

int main(){
    string str1 = "anagram";
    string str2 = "margana";
    cout<< boolalpha<<valid_anagram(str1, str2);
}